src/Microsoft.OData.Edm/Csdl/CsdlReader.cs

```diff
@@ -369,25 +370,25 @@ private List<CsdlModel> LoadAndParseReferencedCsdlFiles(Version mainCsdlVersion)
                 return referencedAstModels;
             }
 
-            foreach (var edmReference in this.edmReferences)
+            foreach (var reference in this.references)
             {
-                if (!edmReference.Includes.Any() && !edmReference.IncludeAnnotations.Any())
+                if (!reference.Includes.Any() && !reference.IncludeAnnotations.Any())
                 {
                     this.RaiseError(EdmErrorCode.ReferenceElementMustContainAtLeastOneIncludeOrIncludeAnnotationsElement, Strings.EdmxParser_InvalidReferenceIncorrectNumberOfIncludes);
                     continue;
                 }
 
-                if (edmReference.Uri != null && (edmReference.Uri.OriginalString.EndsWith(CoreVocabularyConstants.VocabularyUrlSuffix, StringComparison.Ordinal) ||
-                    edmReference.Uri.OriginalString.EndsWith(CapabilitiesVocabularyConstants.VocabularyUrlSuffix, StringComparison.Ordinal) ||
-                    edmReference.Uri.OriginalString.EndsWith("/Org.OData.Authorization.V1.xml", StringComparison.Ordinal) ||
-                    edmReference.Uri.OriginalString.EndsWith("/Org.OData.Validation.V1.xml", StringComparison.Ordinal) ||
-                    edmReference.Uri.OriginalString.EndsWith("/Org.OData.Community.V1.xml", StringComparison.Ordinal) ||
-                    edmReference.Uri.OriginalString.EndsWith(AlternateKeysVocabularyConstants.VocabularyUrlSuffix, StringComparison.Ordinal)))
+                if (reference.Uri != null && (reference.Uri.EndsWith(CoreVocabularyConstants.VocabularyUrlSuffix, StringComparison.Ordinal) ||

```

Comment:
duplicate code to CsdlReader.Json.cs.
 Should this be factored into a helper method like bool ShouldIncludeReferenceInModel(URI uri)?